src/state/pools/index.ts

```diff
@@ -134,13 +134,12 @@ export const fetchCakePoolUserDataAsync = (account: string) => async (dispatch)
   )
 }
 
-export const fetchPoolsPublicDataAsync = () => async (dispatch, getState) => {
+export const fetchPoolsPublicDataAsync = (currentBlockNumber: number) => async (dispatch, getState) => {
   try {
     const blockLimits = await fetchPoolsBlockLimits()
     const totalStakings = await fetchPoolsTotalStaking()
     const profileRequirements = await fetchPoolsProfileRequirement()
-    let currentBlock = getState().block?.currentBlock
-
+    let currentBlock = currentBlockNumber
     if (!currentBlock) {
       currentBlock = await simpleRpcProvider.getBlockNumber()
```

Comment:
Ideally, we should only use `currentBlock` from swr as a source of truth. I think we could remove `initialBlock` by using `currentBlock` as condition. For example
```
if (!currentBlock) {
  fetchPoolsPublicDataAsync(currentBlock)
}
```

Reply:
How do you get `currentBlock` if no `currentBlock` 😆

Reply:
Fixed it. Wrong logic. :)

Reply:
Got it. Thanks

Reply:
But I can remove that fallback and rely on the parameter. What do you think ?

Reply:
Let's keep it. The block number is `0` at first time